website/server/controllers/api-v3/tasks.js

```diff
@@ -166,7 +166,7 @@ api.getTask = {
 
     if (!task) {
       throw new NotFound(res.t('taskNotFound'));
-    } else if (!task.userId) { // If the task belongs to a challenge make sure the user has rights
+    } else if (task.challenge.id && !task.userId) { // If the task belongs to a challenge make sure the user has rights
```

Comment:
Maybe pull this boolean logic into a helper function that takes a task object?  It seems you make this check often enough that it would be beneficial.